src/hotspot/share/gc/shenandoah/shenandoahCollectionSet.cpp

```diff
@@ -87,6 +87,20 @@ void ShenandoahCollectionSet::add_region(ShenandoahHeapRegion* r) {
   assert(!r->is_humongous(), "Only add regular regions to the collection set");
 
   _cset_map[r->index()] = 1;
+
+  if (ShenandoahHeap::heap()->mode()->is_generational()) {
+    if (ShenandoahHeap::heap()->is_aging_cycle()) {
+      r->decrement_age();
```

Comment:
Why are we decreasing the age here? Putting the region in the collection set will call `make_cset` on the region which sets the age to zero.

Reply:
Sorry.  Thanks for catching this.  This was my "compensation" that I had inserted when I was trying to increment all ages right after final mark.  When I backed out that change, I forgot to remove this.  Gone now.